frontend/src/app/main/ui/workspace/viewport/widgets.cljs

```diff
@@ -152,33 +189,47 @@
-            [:use {:href "#icon-boards-thumbnail"}]
-
-            grid-edition?
-            [:use {:href "#icon-grid"}]
-
-            main-instance?
-            [:use {:href "#icon-component"}])])
-
-
-       [:foreignObject {:x text-pos-x
-                        :y -11
-                        :width (max 0 (- text-width text-pos-x))
-                        :height 20
-                        :class (stl/css :workspace-frame-label-wrapper)
-                        :style {:fill color}
-                        :visibility (if show-artboard-names? "visible" "hidden")}
-        [:div {:class (stl/css :workspace-frame-label)
-               :style {:color color}
-               :on-pointer-down on-pointer-down
-               :on-double-click on-double-click
-               :on-context-menu on-context-menu
-               :on-pointer-enter on-pointer-enter
-               :on-pointer-leave on-pointer-leave}
-         (if show-id?
-           (dm/str (dm/str (:id frame)) " - " (:name frame))
-           (:name frame))]]])))
+            (:use-for-thumbnail frame) [:use {:href "#icon-boards-thumbnail"}]
+            grid-edition? [:use {:href "#icon-grid"}]
+            main-instance? [:use {:href "#icon-component"}])])
+
+       (if ^boolean edition?
+           ;; Case when edition? is true
+         [:foreignObject {:x text-pos-x
+                          :y -15
+                          :width (max 0 (- text-width text-pos-x))
+                          :height 22
+                          :class (stl/css :workspace-frame-label-wrapper)
+                          :style {:fill color}
+                          :visibility (if show-artboard-names? "visible" "hidden")}
+          [:input {:type "text"
+                   :class (stl/css :workspace-frame-label
+                                   :element-name-input)
+                   :style {:color color}
+                   :auto-focus true
+                   :on-key-down on-key-down
+                   :ref ref
+                   :default-value (:name frame)
+                   :on-blur accept-edit}]]
+           ;; Case when edition? is false
+         [:foreignObject {:x text-pos-x
+                          :y -11
+                          :width (max 0 (- text-width text-pos-x))
+                          :height 20
+                          :class (stl/css :workspace-frame-label-wrapper)
+                          :style {:fill color}
+                          :visibility (if show-artboard-names? "visible" "hidden")}
+          [:div {:class (stl/css :workspace-frame-label)
+                 :style {:color color}
+                 :ref ref
+                 :on-pointer-down on-pointer-down
+                 :on-double-click start-edit
+                 :on-context-menu on-context-menu
+                 :on-pointer-enter on-pointer-enter
+                 :on-pointer-leave on-pointer-leave}
+           (if show-id?
+             ((dm/str (:id frame)) " - " (:name frame))
```

Comment:
I think this is incorrect, you are creating and string and calling it. The extra `()` should be removed

Reply:
looks like there's a lint error if I remove the extra (), it's old code that was already there so I don't know if the changes are breaking anything

Reply:
It should be `(dm/str (:id frame) " - " (:name frame))`